packages/typography/src/Typography.tsx

```diff
@@ -102,29 +102,50 @@ Subtitle.displayName = 'Subtitle';
  */
 
 type BodyFontWeight = 'regular' | 'medium';
-type BodyProps = HTMLElementProps<'p'> & {
+type BodyProps<T extends keyof JSX.IntrinsicElements> = HTMLElementProps<T> & {
   /**
    * font-weight applied to typography element
    * default: `regular`
    */
   weight?: BodyFontWeight;
-  as?: keyof JSX.IntrinsicElements;
+  as?: T;
 };
 
-const Body: ExtendableBox<BodyProps, 'p'> = ({
+function Body<T extends keyof JSX.IntrinsicElements>({
   className,
   weight = 'regular',
-  as = 'p',
+  as = 'p' as T,
   ...rest
-}: BodyProps) => {
+}: BodyProps<T>) {
   const size = useBaseFontSize();
   const body = size === 16 ? typeScale2 : typeScale1;
+  const fontWeights: {
+    [key: string]: {
+      [key in BodyFontWeight]: number;
+    };
+  } = {
+    default: {
+      regular: 400,
+      medium: 500,
+    },
+    strong: {
+      regular: 700,
+      medium: 800,
+    },
+    b: {
+      regular: 700,
+      medium: 800,
+    },
```

Comment:
Not really sure it's worth separating `strong` & `b`

Reply:
I agree. I left it initially to demonstrate how we can have 1-to-1 mapping, but it's not that important now we're taking this middle ground route.